src/gui/app.cpp

```diff
@@ -13,6 +13,65 @@
 
 namespace NeovimQt {
 
+struct ConnectorInitArgs {
+	enum Type {
+		Embed,
+		Server,
+		Spawn,
+		Default
+	};
+
+	const Type type;
+	const int timeout;
+	const QString server;
+	const QString nvim;
+	const QStringList positionalArgs;
+	const QStringList neovimArgs;
```

Comment:
Let's provide a CTOR and delete the default CTOR.

```C++
ConnectorInitArgs(const QCommandLineParser& parser, const QStringList& neovimArgs) noexcept
{
..
}
```

Use the prefix `m_` for member variables.

Consider a class, and moving this to its own file if complexity increases. A struct here is fine for now.

Reply:
I was intending this to be a very simple  data structure to hold on to the initialization invariants. If I add a ctor that takes a parser, I would need to create another ctor later when a different configuration is used (e.g multiple windows). 

How about a function that returns this struct from a parser?

Reply:
A few comments:
1. I'd strongly encourage a class w/ getters/setters. A struct is harder to expand in the future. If you prefer a simple struct, that's fine.
2. A CTOR will not increase the complexity. Multiple CTORs is fine. A CTOR is the correct C++ mechanism for what you're doing.
3. Yeah, I suppose `m_` is a little awkward for public struct members. Feel free to ignore `m_`.

> How about a function that returns this struct from a parser?

Well, that's a CTOR... Right?

Writing a CTOR instead of a function returning the object allows the compiler to perform additional optimizations. Your code is all great, it simply needs to be moved into the CTOR.

```
struct struct ConnectorInitArgs
{
    static Type getConnectorType(const QCommandLineParser& parser) noexcept;

    ConnectorInitArgs() noexcept = delete;

    ConnectorInitArgs(const QCommandLineParser& parser, const QStringList& neovimArgs) noexcept
        : server{ ... }
	, nvim{ ... }
	, positionalArgs{ ... }
	, neovimArgs{ ... }
    { ... }
    ...
```

A CTOR helps prevent contributors in the future from introducing bugs.

Ex) It is very easy to accidentally switch the ordering of variables and break your code.